src/http/client.rs

```diff
@@ -75,9 +71,12 @@ pub struct EspHttpClientConfiguration {
     pub follow_redirects_policy: FollowRedirectsPolicy,
 }
 
+trait EventHandler = Fn(&esp_http_client_event_t) + 'static;
```

Comment:
Is this a [trait alias](https://doc.rust-lang.org/beta/unstable-book/language-features/trait-alias.html)?
Was this stabilized recently and I just missed that (wondering how it even works without `#![feature(trait_alias)]`)? If not stabilized, I wonder if we want it (I already have tons of problems with GATs + HRTBs) so I don't want to make the whole thing even less stable in terms of compiler crashes.

Reply:
Oh shoot, it is, and I didn't intend to introduce any unstable features (I don't think it has been stabilized yet?!). I'll remove the alias ASAP.

Reply:
I suspect this code is excluded by some feature flag that is not used in the check build?

Reply:
Yes. you need a `--features experimental` of course. This is BTW why the CI did not catch it. I was lazy and did not implement (yet) a run with that one ^^^; neither a run with no_std. We need both actually, to catch all errors.

Reply:
By the way, I DID introduce one unstable feature of course: GATs. But my hope is it is really close to being stabilized, and the ESP-IDF story is anyway a nightly-only thing (because `cargo -Z build-std` is for now a nightly-only thing) but I'm not keen on introducing more unstable features.